### PR TITLE
Add support for vertex color attributes to `AccessorUtility`

### PR DESCRIPTION
**Pull request by lilleyse**
## Description

Adds support for vertex color attributes to `AccessorUtility`.

`ColorFromAccessor` always returns a `glm::dvec4`. If the underlying type is `VEC3` the alpha channel is set to 1.0.

image

## Author checklist

- [x] I have submitted a [Contributor License Agreement](https://github.com/CesiumGS/community/tree/main/CLAs) (only needed once).
- [x] I have done a full self-review of my code.
- [x] I have updated `CHANGES.md` with a short summary of my change (for user-facing changes).
- [x] I have added or updated unit tests to ensure consistent code coverage as necessary.
- [x] I have updated the documentation as necessary.

## Reviewer checklist

Thank you for taking the time to review this PR. By approving a PR you are taking as much responsibility for these changes as the author.

As you review, please go through the checklist below:

- [ ] Review and run all parts of the test plan on this branch and verify it matches expectations.
    - [ ] If the issue is a bug please make sure you can reproduce the bug in the main branch and then checkout this branch to make sure it actually solved the issue.
- [ ] Review the code and make sure you do not have any remaining questions or concerns. You should understand the code change and the chosen approach. If you are not confident or have doubts about the code, please do not hesitate to ask questions.
    - [ ] Code should follow the [C++ Style Guide](https://cesium.com/learn/cesium-native/ref-doc/style-guide.html)
- [ ] Review the unit tests and make sure there are no missing tests or edge cases.
- [ ] Review documentation changes and updates to `CHANGES.md` to make sure they accurately cover the work in this PR.
- [ ] Verify that the [Contributor License Agreement](https://github.com/CesiumGS/community/tree/main/CLAs) has been submitted, if needed.

### CODE REVIEW

File: CesiumGltf/include/CesiumGltf/AccessorUtility.h [hunk 1]
```diff
@@ -571,4 +575,93 @@ struct QuaternionFromAccessor {
...
+
+/**
+ * Retrieves an accessor view for the color attribute from the given glTF
+ * primitive and model. This verifies that the accessor is of a valid type. If
+ * not, the returned accessor view will be invalid.
+ */
+ColorAccessorType getColorAccessorView(
+    const Model& model,
+    const MeshPrimitive& primitive,
+    int32_t colorSetIndex);
```
Comment (j9liu) [inline CesiumGltf/include/CesiumGltf/AccessorUtility.h]: Maybe mention what `colorSetIndex` refers to?

This made me realize that these comments should have been using the `@brief`, etc. tags... if you can add them for this function, at least, that'd be great :')

File: CesiumGltf/src/AccessorUtility.cpp [hunk 2]
```diff
@@ -258,4 +258,68 @@ getQuaternionAccessorView(const Model& model, int32_t accessorIndex) {
...
+      } else {
+        // Invalid accessor, UNSIGNED_SHORT must be normalized
+        return ColorAccessorType();
+      }
+    case Accessor::ComponentType::FLOAT:
+      return AccessorView<AccessorTypes::VEC3<float>>(model, *pAccessor);
+    default:
+      return ColorAccessorType();
+    }
+  } else if (pAccessor->type == Accessor::Type::VEC4) {
```
Comment (j9liu) [inline CesiumGltf/src/AccessorUtility.cpp]: **NIT**: Since the first `if` body will always `return`, it reads nicer IMO to break this into a separate `if` statement :')

```suggestion
  }
  
  if (pAccessor->type == Accessor::Type::VEC4) {
```

File: CesiumGltf/test/TestAccessorUtility.cpp [hunk 3]
```diff
@@ -1640,3 +1640,395 @@ TEST_CASE("Test QuaternionFromAccessor") {
...
+      accessor.bufferView = int32_t(model.bufferViews.size() - 1);
+      accessor.componentType = componentType;
+      accessor.type = type;
+      accessor.count = int64_t(colors.size());
+      accessor.normalized = normalized;
+
+      std::string name = fmt::format("COLOR_{}", setIndex);
+      primitive.attributes.emplace(name, int32_t(model.accessors.size() - 1));
+    }
+  };
```
Comment (j9liu) [inline CesiumGltf/test/TestAccessorUtility.cpp]: Is this identical to the function in `getColorAccessorView`? If so, it would save some space if we broke it out as a helper function in anonymous `namespace`...

**Review by j9liu — COMMENTED** [review]
Thanks @lilleyse ! Looks great, I just have a few small comments before merge.

**Comment by lilleyse** [comment]
@j9liu Thanks! Ready for another review.

**Review by j9liu — APPROVED** [review]
Thanks @lilleyse !
